pkg/resource/resource_goal.go

```diff
@@ -36,13 +36,14 @@ type Goal struct {
 	AdditionalSecretOutputs []PropertyKey         // outputs that should always be treated as secrets.
 	Aliases                 []URN                 // additional URNs that should be aliased to this resource.
 	ID                      ID                    // the expected ID of the resource, if any.
+	CustomTimeouts          *CustomTimeouts       // an optional config object for resource options
```

Comment:
Given that we're treating a timeout of zero seconds as "just use the default", should we go ahead and make this a value rather than a pointer? It'd be nice to avoid the `nil` checks, and I don't think using a pointer adds any beneficial semantics.

Reply:
done!